src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -1861,13 +1861,14 @@ void SystemDictionaryShared::record_linking_constraint(Symbol* name, InstanceKla
     return;
   }
 
-  if (THREAD->is_VM_thread()) {
-    assert(DynamicDumpSharedSpaces, "must be");
+  if (DynamicDumpSharedSpaces && Thread::current()->is_VM_thread()) {
```

Comment:
This is still a functional change. If there is a bug and we get here in the VMThread when DynamicDumpSharedSpaces is not set then we will no longer immediately return.

Reply:
That's fine. In debug build, we will assert both before/after this change. In product build, after this change, some duplicated info will be added via info->record_linking_constraint but it won't hurt.

Reply:
Okay. Thanks for clarifying.

Reply:
Thanks for looking carefully at this.